dev-docs/RFCs/v8.0/ui-module-rfc.md

```diff
@@ -0,0 +1,137 @@
+# UI Module RFC
+
+* **Author**: Xiaoji Chen
+* **Date**: December 2022
+* **Status**: **Pending Review**
+
+## Overview
+
+Visualization applications often require some chrome around the WebGL canvas to offer better user experience. While it is possible for every app to implement their own UI components that interact with deck.gl, it is no trivial effort to implement them well. 
+
+This RFC proposes to add a new module that packages common UI components that seamlessly integrate with deck.gl, including but not limited to:
+
+- Marker: HTML element pinned to a location in the world space
+- Navigation: buttons to zoom / rotate / reset the viewport
+- Fullscreen: button to enter / exit fullscreen mode
+- Geolocate: button to track the user's location using the browser's Geolocation API
+- Legend: visual comparison of sizes, colors etc. corresponding to the rendered layers and viewport
```

Comment:
A layer legend like the one in kepler.gl would be very useful. It'd be very challenging to represent every visual prop in every layer type, plus it could be an expensive operation to impute metrics from the data. I wonder how to balance this... perhaps users should be expected supply values themselves?

A world scale legend for MapView's zoom level.

Reply:
Other very useful components: 
- A compass rose - could offer both a 2D rose or a 3D / tilting rose
- Distance ruler - Could support 3D views by showing distance scale at the place where it is rendered.
- A simple temporal playback widget has been a top request from pydeck